src/main/java/labs/introtoprogramming/lab5/gui/SceneRendererWindow.java

```diff
@@ -0,0 +1,72 @@
+package labs.introtoprogramming.lab5.gui;
+
+import java.util.Timer;
+import java.util.TimerTask;
+import labs.introtoprogramming.lab5.exception.NoCameraException;
+import labs.introtoprogramming.lab5.scene.BasicRaytracingRender;
+import labs.introtoprogramming.lab5.scene.Scene;
+import labs.introtoprogramming.lab5.scene.SceneRender;
+
+public class SceneRendererWindow extends Window {
+
+  private static final String UPDATE_LOOP_TIMER_NAME = "UpdateLoop";
+  private static final int UPDATE_LOOP_DELAY = 100; // ms
+  private static final int TARGET_FPS = 60;
+  private static final int UPDATE_RATE = 1000/TARGET_FPS;
+  private static final boolean DEFAULT_CAMERA_CONTROLLABLE = false;
+
+  private SceneRender render;
+  private Scene scene;
+  private MouseController mouseController;
+  private Input input;
+
+  public SceneRendererWindow(Scene scene) {
+    this(new BasicRaytracingRender(scene.getCamera().orElseThrow(NoCameraException::new).raster()), scene);
```

Comment:
Maybe is it better to refactor SceneRender to only accept scene, if raster can be obtain from camera now? Or is it out of place for this PR?

Reply:
Multiple constructors increase flexibility. There should be a way to set custom scene render.

Reply:
Other constructors allow more flexibility, for example to set custom scene render.

Reply:
I am talking about basicRaytracingRender. Maybe is it better to add constructor that only accept scene as argument and take raster from camera?

Reply:
Oh, right then.